routers/repo/branch.go

```diff
@@ -222,32 +231,23 @@ func loadBranches(ctx *context.Context, page, pageSize int) ([]*Branch, int) {
 	repoIDToGitRepo := map[int64]*git.Repository{}
 	repoIDToGitRepo[ctx.Repo.Repository.ID] = ctx.Repo.GitRepo
 
-	var totalNumOfBranches = len(rawBranches)
-	var startIndex = (page - 1) * pageSize
-	if startIndex > totalNumOfBranches {
-		startIndex = totalNumOfBranches - 1
-	}
-	var endIndex = startIndex + pageSize
-	if endIndex > totalNumOfBranches {
-		endIndex = totalNumOfBranches - 1
-	}
-
 	var branches []*Branch
-	for i := startIndex; i < endIndex; i++ {
+	for i := range rawBranches {
+		if rawBranches[i].Name == defaultBranch.Name {
+			// Skip default branch
+			continue
+		}
+
 		var branch = loadOneBranch(ctx, rawBranches[i], protectedBranches, repoIDToRepo, repoIDToGitRepo)
 		if branch == nil {
 			return nil, 0
 		}
 
-		if branch.Name == ctx.Repo.Repository.DefaultBranch {
-			// Skip default branch
-			continue
-		}
-
 		branches = append(branches, branch)
 	}
 
 	// Always add the default branch
+	log.Info("loadOneBranch: load default: '%s'", defaultBranch.Name)
```

Comment:
This should at Debug at most - perhaps even trace or not at all.

```suggestion
	log.Debug("loadOneBranch: load default: '%s'", defaultBranch.Name)
```